### DISCUSSION TITLE
Should I be able to use a notes `created` field as the input to the `--created-before` flag?

### DISCUSSION THREAD

**Discussion by mhanberg** [zk-org/zk#120] [Help]
I am writing some code to help with my dungeons and dragons notes, and I want to have a keymap to insert the link to the previous note.

I am able to get the current note with

```lua
-- in zk.nvim
local get = function(file_path, fields)
	local default_fields = { "title", "absPath" }
	local err, result = zk.list({ select = fields or default_fields, limit = 1, hrefs = { file_path } })

  return err, result[1]
end

-- in my dotfiles
dnd.previous_session = function()
  local err, current_session = zk.get(vim.fn.expand("%"), { "title", "absPath", "created" })

  if err then
    vim.api.nvim_err_write("[zk] Error fetching the current session's note: " .. vim.inspect(err))
    return
  end

  local previous_session_err, last_session = zk.list({
    select = { "title", "absPath" },
    excludeHref = { current_session.absPath },
    createdBefore = current_session.created,
    limit = 1,
    sort = {"created"},
  })

  if previous_session_err then
    vim.api.nvim_err_write("[zk] Error fetching the previous session's note: " .. vim.inspect(previous_session_err))
    return
  end

  return last_session[1]
end
```

This will error out with

```
[zk] Error fetching the previous session's note: {
  code = -32600,
  message = '\nparse error near PegText (line 1 symbol 1 - line 1 symbol 5):\n"2021"\n',
  <metatable> = {
    __tostring = <function 1>
  }
```

I assume this is because the formatting of the date is not being parsed correctly.

Am I supposed to be able to do this?

My note frontmatter looks like this.
```
---
title: DnD - Monday, November 29, 2021
tags: [dnd, session]
date: Monday, November 29, 2021
---
```

**Comment by mickael-menu** [comment]
Yes I think you should be able to do that. `--created-before` used to take only "natural human dates" as format, but since we can now call it programmatically, it makes sense to support RFC3339 as well (which is returned in the LSP JSON).

Could you test the fix in the `parse-rfc3339-date` branch and see if it gives the expected results for you?

**Reply by mhanberg** [reply]
is that branch rebased on main?

**Reply by mhanberg** [reply]
oh i see that's a new branch, i'll try it out

**Reply by mhanberg** [reply]
That branch is working for me!

**Reply by mickael-menu** [reply]
Okay, thanks! I'm planning on doing a proper 0.9.0 this week-end with all the recent changes.
